server/graphql/core/src/loader/document.rs

```diff
@@ -39,15 +39,31 @@ impl Loader<DocumentLoaderInput> for DocumentLoader {
         let jobs = doc_names_by_store(names);
         let mut out = HashMap::new();
         for (store_id, doc_names) in jobs {
-            let result = self.service_provider.document_service.get_documents(
-                &ctx,
-                Some(
-                    DocumentFilter::new()
-                        .name(StringFilter::equal_any(doc_names.into_iter().collect())),
-                ),
-                None,
-            )?;
-            for doc in result {
+            let result = self
+                .service_provider
+                .document_service
+                .documents(
+                    &ctx,
+                    None,
+                    Some(
+                        DocumentFilter::new()
+                            .name(StringFilter::equal_any(doc_names.into_iter().collect())),
+                    ),
+                    None,
+                    None,
+                )
+                .map_err(|err| match err {
+                    ListError::DatabaseError(err) => err,
+                    ListError::LimitBelowMin(_) => RepositoryError::DBError {
```

Comment:
Would these errors be BadUserInput? 🤔

Reply:
Good catch. I was a bit sloppy here, this error should never happen because I don't specify a pagination nor the user can specify one.  I named it internal error now.

General note, we should move the pagination min/max check into the graphql layer. We shouldn't restrict us how we can use the service here because there are use-cases where we just want to get all data and it hard to define an upper limit. However, we want to restrict external users to load too much data at a time, e.g. because this could cause us to run out of memory -> crash the server. The ListError would just become a Repository error which would also simplify some things.